Tasks/SqlDacpacDeploymentOnMachineGroup/GenerateSqlBatchFiles.ps1

```diff
@@ -0,0 +1,215 @@
+        if ($lineSubstring -match " GO")
+        {
+            return $line.Substring(0, [int]$length - 3), $null
+        }
+    }
+       
+    #Send the line as is       
+    return $line, $null
+}
+
+function Test-IsSplitRequired
+{
+    param
+    (
+        [string]$line
+    )
+
+    $line = $line.Trim()
+    $length = $line.Length
+    $commentsIndex = $line.IndexOf("--")
+
+    #Don't split if entire line is a comment
+    if ($commentsIndex -eq 0)
+    {
+        return $false
+    }
+
+
+    #Split if the line has only GO
+    if ($line -contains "GO")
+    {
+        return $true
+    }
+
+    #Split if line starts with GO
+    #This avoids 'select * from indigo join table2' even though there is matching pattern 'go '
+    if ($length -ge 3)
+    {
+        $lineSubstring = $line.Substring(0, 3)
+        if ($lineSubstring -match "GO ")
+        {
+            return $true
+        }
+    }
+    
+    #Split line if it has a GO in between like 'select * from table GO create procedure'
+    $goIndex = $line.IndexOf(" GO ", [System.StringComparison]::CurrentCultureIgnoreCase)
+    if (($commentsIndex -ne -1) -and ($goIndex -ne -1) -and ($commentsIndex -lt $goIndex))
+    {
+        return $false
+    }
+    if ($goIndex -ge 0)
+    {
+        return $true
+    }
+
+    #Split if line ends with GO
+    #This avoids 'select * from indigo' even if there is a matching pattern ending with 'go'
+    $lastCharsIndex = [int]$length - 3
+    if (($commentsIndex -ne -1) -and ($lastCharsIndex -ge 0) -and ($commentsIndex -lt $lastCharsIndex))
+    {
+        return $false
+    }
+    if ($length -ge 3)      
+    {
+        $lineSubstring = $line.Substring($lastCharsIndex, 3)
+        if ($lineSubstring -match " GO")
+        {
+            return $true
+        }
+    }
+     
+    return $false
+}
+
+function Write-BatchFile
+{
+    param
+    (
+        [string]$fileName,
+        [string]$line,
+        [boolean]$newBatch,
+        [string]$filesCreated
+    )
+    
+    Add-Content -path $fileName -value $line
+
+    if ($newBatch -eq $true)
+    {
+        $filesCreated = $filesCreated + $fileName + "; "
+    }
+
+    $newBatch = $false
+
+    return $newBatch, $filesCreated
+}
+
+function Get-NewBatchFileName
+{
+   param
+    (
+        [string]$destPath,
+        [string]$batch,
+        [string]$batchIndex,
+        [string]$sqlFileName,
+        [string]$ext
+    )
+    
+    $fileName = "{0}\{1}.{2}.{3}.{4}" -f ($destPath, $batch, $batchIndex, $sqlFileName, $ext)
+    return $fileName
+}
+
+function Create-BatchFilesForSqlFile 
+{
+    param
+    (
+        [string]$sqlFilePath,
+        [string]$destPath,
+        [string]$batch
+    )
+    
+    $batchIndex = 1
+    $sqlFileName = Split-Path $sqlFilePath -Leaf
+    $reader = new-object System.IO.StreamReader($sqlFilePath)  
+    $ext = "sql"
+    $batchFileName = Get-NewBatchFileName -destPath $destPath -batch $batch -batchIndex $batchIndex -sqlFileName $sqlFileName -ext $ext
+
```

Comment:
The batch file name is generated as 1.1.filename.sql.sql. Should it just be 1.1.filename.sql ?

Reply:
This still works. These are temporary batches. Hence, I can take this if it is really required.